### PR TITLE
Don't inflect meta tag name/type.

### LINKED ISSUE #17910: Under the hood rename of meta name
### Description

This is sneaky change that will break a lot of apps. When creating meta csrfToken is renamed to csrf-token. I don't understand the reason, but when someone wants to name the tag csrfToken, it should stay csrfToken

### CakePHP Version

5.1

### PHP Version

8.3

### PR DESCRIPTION
**Pull request by ADmad**
This avoids BC issues. Closes #17910.

### CODE REVIEW

File: src/View/Helper/HtmlHelper.php [hunk 1]
```diff
@@ -137,7 +137,7 @@ public function meta(array|string $type, array|string|null $content = null, arra
     {
         if (is_string($type)) {
             $types = [
-                'csrfToken' => ['name' => 'csrf-token'],
+                'csrf-token' => ['name' => 'csrf-token'],
```
Comment (markstory) [inline src/View/Helper/HtmlHelper.php:140]: I'm assuming there were applications using `csrfToken` and providing the `content` parameter as well? Couldn't we end up breaking a different set of applications with a rename?

File: src/View/Helper/HtmlHelper.php [hunk 2]
```diff
@@ -156,8 +156,8 @@ public function meta(array|string $type, array|string|null $content = null, arra
                 $types['icon']['link'] = 'favicon.ico';
             }
 
-            if ($type === 'csrfToken') {
-                $types['csrfToken']['content'] = $this->_View->getRequest()->getAttribute('csrfToken');
+            if ($type === 'csrf-token') {
```
Comment (markstory) [inline src/View/Helper/HtmlHelper.php:159]: Perhaps we should only set the token if `content` is null? That would help avoid the problem just as much as renaming the magic key does.

File: src/View/Helper/HtmlHelper.php (hunk 1 above)
Comment (ADmad) [reply src/View/Helper/HtmlHelper.php:140]: > I'm assuming there were applications using csrfToken and providing the content parameter as well?

Yes, the problem was they expected a meta tag with name `csrfToken` but ended up with `csrf-token` instead.

> Couldn't we end up breaking a different set of applications with a rename?

That would be possible only if they were using some other content value than the one we have in the request attribute, but I don't see what good that would do.

This issue right now isn't the value but the name inflection.

File: src/View/Helper/HtmlHelper.php (hunk 2 above)
Comment (ADmad) [reply src/View/Helper/HtmlHelper.php:159]: Yes that's an option too.

Comment (ADmad) [reply src/View/Helper/HtmlHelper.php:159]: In hindsight I should have used  the same magic name as the tag name.

File: src/View/Helper/HtmlHelper.php (hunk 1 above)
Comment (markstory) [reply src/View/Helper/HtmlHelper.php:140]: > That would be possible only if they were using some other content value than the one we have in the request attribute, but I don't see what good that would do.

Me neither, but we've gotten issues for all sorts of changes over the years :shrug:

Comment (nitramxx) [reply src/View/Helper/HtmlHelper.php:140]: FYI, this is the code:

`$this->Html->meta('csrfToken', $this->request->getAttribute('csrfToken'))`

**Pull request merged**